website/docs/developer/research/research-s3-bucket-policies.md

```diff
@@ -0,0 +1,156 @@
+With a default policy of `Governance` mode and an arbitrarily long `Retention period`, data will be protected from accidental deletion by general users.
+
+### Additional Considerations
+
+Note that due to poor API returns, attempting to delete a file that cannot be deleted returns HTTP status code 204. When deletes are added to Orca, effort should be taken to return a proper error/success response to customers.
+
+Due to technically being versioned, care should be taken when uploading as well.
+Files should be rejected if they are already present in a WORM bucket, likely through an object lookup on the bucket.
+
+There is still a race condition if two clients upload the same file simultaneously.
+Checks should be done after a file is uploaded to make sure it is still the only version present.
+If not, the uploaded version should be deleted, and an error message sent back to the client.
+In the event that this mechanic fails, Internal Reconciliation should report errors on multiple versions of the same file within a WORM bucket.
+
+## Reports Bucket
+
+Suggested name: PREFIX-orca-reports
+
+This bucket is relatively simple, as much of the interaction is automated through AWS. This bucket stores S3 Inventory reports generated by AWS, to be pulled in by `get_current_archive_list`.
+Versioning is not used, and in general no changes are required beyond updating the Permissions to the policy below:
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "Cross Account Access",
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "arn:aws:iam::909121343565:root"
+      },
+      "Action": [
+        "s3:GetObject*",
+        "s3:GetBucket*",
+        "s3:ListBucket",
+        "s3:PutObject",
+        "s3:PutObjectAcl",
+        "s3:PutBucketNotification"
+      ],
+      "Resource": [
+        "arn:aws:s3:::PREFIX-orca-glacier-archive-versioned",
+        "arn:aws:s3:::PREFIX-orca-glacier-archive-versioned/*"
+      ]
+    },
+    {
+      "Sid": "Inventory-PREFIX-orca-glacier-archive-versioned",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "s3.amazonaws.com"
+      },
+      "Action": "s3:PutObject",
+      "Resource": "arn:aws:s3:::PREFIX-orca-glacier-archive-versioned/*",
+      "Condition": {
+        "StringEquals": {
+      	  "s3:x-amz-acl": "bucket-owner-full-control",
+      	  "aws:SourceAccount": "782417781503"
+        },
+      	"ArnLike": {
+      	  "aws:SourceArn": "arn:aws:s3:::PREFIX-orca-glacier-archive-versioned"
+      	}
+      }
+    },
+    {
+      "Sid": "Inventory-PREFIX-orca-glacier-archive-worm",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "s3.amazonaws.com"
+      },
+      "Action": "s3:PutObject",
+      "Resource": "arn:aws:s3:::PREFIX-orca-glacier-archive-worm/*",
+      "Condition": {
+        "StringEquals": {
+      	  "s3:x-amz-acl": "bucket-owner-full-control",
```

Comment:
add a note that this will be deprecated in the future and might need some replacement.

Reply:
It is at the end of the section.